src/Rules/Properties/PropertyReflectionFinder.php

```diff
@@ -12,6 +12,50 @@
 class PropertyReflectionFinder
 {
 
+	/**
+	 * @param \PhpParser\Node\Expr\PropertyFetch|\PhpParser\Node\Expr\StaticPropertyFetch $propertyFetch
+	 * @param \PHPStan\Analyser\Scope $scope
+	 * @param array<string, \PHPStan\Type\Constant\ConstantStringType> $newPropertyNamesAndTypes
+	 * @return FoundPropertyReflection[]
+	 */
+	public function findPropertyFromArrayAssignment($propertyFetch, Scope $scope, array $newPropertyNamesAndTypes): array
```

Comment:
I think this method should have this API:

```php
/**
 * @return FoundPropertyReflection[]
 */
public function findPropertyReflectionsFromNode($propertyFetch, Scope $scope): ?array
```

And it should be used everywhere `findPropertyReflectionFromNode` is currently used. And `findPropertyReflectionFromNode` should be marked as deprecated.

It also means that more places will now use foreach instead of working with a single found reflection.

Reply:
done